nifi-nar-bundles/nifi-hubspot-bundle/nifi-hubspot-processors/src/main/java/org/apache/nifi/processors/hubspot/GetHubSpot.java

```diff
@@ -0,0 +1,263 @@
+            } else {
+                getLogger().debug("Empty response when requested HubSpot endpoint: [{}]", endpoint);
+            }
+        } else if (response.statusCode() >= 400) {
+            if (response.statusCode() == TOO_MANY_REQUESTS) {
+                context.yield();
+                throw new ProcessException(String.format("Rate limit exceeded, yielding before retrying request. HTTP %d error for requested URI [%s]", response.statusCode(), uri));
+            } else {
+                getLogger().warn("HTTP {} error for requested URI [{}]", response.statusCode(), uri);
+            }
+        }
+    }
+
+    private OutputStreamCallback parseHttpResponse(ProcessContext context, String endpoint, StateMap state, HttpResponseEntity response, AtomicInteger objectCountHolder) {
+        return out -> {
+            try (JsonParser jsonParser = JSON_FACTORY.createParser(response.body());
+                 final JsonGenerator jsonGenerator = JSON_FACTORY.createGenerator(out, JsonEncoding.UTF8)) {
+                while (jsonParser.nextToken() != null) {
+                    if (jsonParser.getCurrentToken() == JsonToken.FIELD_NAME && jsonParser.getCurrentName()
+                            .equals("results")) {
+                        jsonParser.nextToken();
+                        jsonGenerator.copyCurrentStructure(jsonParser);
+                        objectCountHolder.incrementAndGet();
+                    }
+                    String fieldName = jsonParser.getCurrentName();
+                    if (CURSOR_PARAMETER.equals(fieldName)) {
+                        jsonParser.nextToken();
+                        Map<String, String> newStateMap = new HashMap<>(state.toMap());
+                        newStateMap.put(endpoint, jsonParser.getText());
+                        updateState(context, newStateMap);
+                        break;
+                    }
+                }
+            }
+        };
+    }
+
+    HttpUriBuilder getBaseUri(final ProcessContext context) {
+        final String path = context.getProperty(CRM_ENDPOINT).getValue();
+        return webClientServiceProvider.getHttpUriBuilder()
+                .scheme(HTTPS)
+                .host(API_BASE_URI)
+                .encodedPath(path);
+    }
+
+    private HttpResponseEntity getHttpResponseEntity(final String accessToken, final URI uri) {
+        return webClientServiceProvider.getWebClientService()
+                .get()
+                .uri(uri)
+                .header("Authorization", "Bearer " + accessToken)
+                .retrieve();
+    }
+
+    private URI createUri(final ProcessContext context, final StateMap state) {
+        final String path = context.getProperty(CRM_ENDPOINT).getValue();
+        final HttpUriBuilder uriBuilder = getBaseUri(context);
+
+        final boolean isLimitSet = context.getProperty(LIMIT).isSet();
+        if (isLimitSet) {
+            final String limit = context.getProperty(LIMIT).getValue();
```

Comment:
Since `LIMIT` supports EL shouldn't `PropertyValue.evaluateAttributeExpressions()` be called before getting the value?